frontend/heir/mlir_emitter.py

```diff
@@ -514,9 +523,12 @@ def emit_assign(self, assign):
             )
           target_ssa = self.get_or_create_name(assign.target)
           ssa_id = self.get_or_create_name(assign.value.args[0])
+
+          # Construct an instance of the MLIR type in question
+          mlir_type_instance = global_()
```

Comment:
This isn't super pretty, but seems like a great tradeoff for making the matching code so much nicer!